fiona/collection.py

```diff
@@ -427,6 +430,30 @@ def __del__(self):
         self.close()
 
 
+def _get_valid_geom_types(schema, driver):
+    schema_geom_type = schema["geometry"]
+    if isinstance(schema_geom_type, string_types) or schema_geom_type is None:
+        schema_geom_type = (schema_geom_type,)
+    valid_types = set()
+    for geom_type in schema_geom_type:
+        valid_types.add(str(geom_type).lstrip("3D "))
+
+    if "Unknown" in valid_types:
```

Comment:
I think "Any" might be better than "Unknown".

Reply:
The reason for using "Unknown" is that this is what GDAL uses when it opens a multi-type layer. I think we need to support it, so that you can reuse the schema when reading from one layer directly into another - I'll add a test specifically for this. I don't mind adding an alias for this for writing layers. "Any" is my preference.

Reply:
Ah, okay, I see it does. I quibble with GDAL's semantics all the time. Consider the two statements:

1. Create a new layer to store features with an *unknown* geometry type
2. Create a new later to store features with *any* geometry type

I think *any* suits the writing situation better and *unknown* suits the reading situation better, if you see what I mean. I think we should choose one and follow GDAL here. "unknown" it is.